### ISSUE TITLE
Some test scenario are randomly failing with latest master

### ISSUE THREAD

**Issue by steweg** [strongswan/strongswan#592]
I was using Ubuntu 20.04, but the problem was happening during pull requests and was happening on Visual Studio 2019 build within AppVeyor.

One of the thread tests was randomly failing. I happened during pull requests and I fixed it by just reordering 1 line within the patch and force-pushed it. It apparently could not affect the failing test. 

Try to create any pull request. As it seems to be random in time, maybe multiple pull request might be needed.

**Expected behavior**
No such failing tests will be happening

**Logs/Backtraces**
See AppVeyor

**Additional context**
N/A

**Comment by Thermi** [comment]
That's just AppVeyor and Windows for you. Some parts of the test suite are non-deterministic on there because Windows is ... weird.

**Comment by steweg** [comment]
I agree, but the problem is that I cannot simply "retrigger" the build if it fails in order to have correct pull requests. Therefore I am forced to amend the commits etc to fix that. Would be nice to either have option to retrigger the tests in such cases

**Comment by Thermi** [comment]
Same for me brother, same for me.
